src/main.ts

```diff
@@ -36,15 +36,61 @@ async function dpkg_list(): Promise<Set<string>> {
   return ret
 }
 
+function semver_max(a: string, b: string): string {
+  const re = /[.-]/
+  var pa = a.split(re)
+  var pb = b.split(re)
+  for (var i = 0; ; i++) {
+    var na = Number(pa[i])
+    var nb = Number(pb[i])
+    if (na > nb) return a
+    if (nb > na) return b
+    if (!isNaN(na) && isNaN(nb)) return a
+    if (isNaN(na) && !isNaN(nb)) return b
+    if (isNaN(na) && isNaN(nb)) return pa[i] >= pb[i] ? a : b
+  }
+}
+
 async function run_linux(): Promise<void> {
   try {
     const httpc = new httpm.HttpClient('httpc')
-    const t_version = core.getInput('tarantool-version', {required: true})
     const lsb_release = await capture('lsb_release -c -s', {silent: true})
     const cache_dir = 'cache-tarantool'
-    const cache_key =
-      core.getInput('cache-key') ||
-      `tarantool-setup-${t_version}-${lsb_release}`
+    let cache_key = core.getInput('cache-key')
+
+    const baseUrl =
+      'https://download.tarantool.org/tarantool/release/' +
+      core.getInput('tarantool-version', {required: true})
+
+    core.startGroup('Checking latest tarantool version')
+    const latest_version = await httpc
+      .get(baseUrl + `/ubuntu/dists/${lsb_release}/main/binary-amd64/Packages`)
+      .then(response => {
+        if (response.message.statusCode !== 200) {
+          throw new Error(`server replied ${response.message.statusCode}`)
+        }
+        return response.readBody()
+      })
+      .then(output => {
+        let ret = ''
+        output
+          .split('\n\n')
+          .filter(paragraph => paragraph.startsWith('Package: tarantool\n'))
```

Comment:
I had doubts regarding whether `Package` field is guaranteed to be the first. However it appears that it is guaranteed.

> Each paragraph shall begin with a "Package" field. Clients may also accept files where this is not the case. 

https://wiki.debian.org/DebianRepository/Format#A.22Packages.22_Indices